internal/config/config.go

```diff
@@ -352,6 +352,7 @@ type RoutingConfig struct {
 	BanWindow       *time.Duration `yaml:"banWindow"`
 	MaxBlocksBehind int            `yaml:"maxBlocksBehind"`
 	IsInitialized   bool
+	IsCheckEnabled  bool
```

Comment:
nit: should it be called `IsEnabled` here?  Since the notion of `check` doesn't exist in `Config`